Sources/Roboflow/Classes/RFClassificationModel.swift

```diff
@@ -0,0 +1,285 @@
+                completion(nil, NSError(domain: "RFClassificationModel", code: 2, userInfo: [NSLocalizedDescriptionKey: "Could not find model input name"]))
+                return
+            }
+            
+            
+            // Get expected image dimensions
+            let inputImage: MLFeatureValue
+            if let imageConstraint = inputDescription.imageConstraint {
+                let targetWidth = Int(imageConstraint.pixelsWide)
+                let targetHeight = Int(imageConstraint.pixelsHigh)
+                
+                // Create a resized pixel buffer
+                let resizedBuffer = resizePixelBuffer(buffer, targetWidth: targetWidth, targetHeight: targetHeight)
+                inputImage = MLFeatureValue(pixelBuffer: resizedBuffer ?? buffer)
+            } else {
+                // Use original buffer if no constraints
+                inputImage = MLFeatureValue(pixelBuffer: buffer)
+            }
+            
+            let inputDict = [inputName: inputImage]
+            let inputProvider = try MLDictionaryFeatureProvider(dictionary: inputDict)
+            
+            // Run prediction directly with CoreML
+            let prediction = try mlModel.prediction(from: inputProvider)
+            
+            // Get the actual output name from model description
+            guard let outputName = mlModel.modelDescription.outputDescriptionsByName.keys.first,
+                  let output = prediction.featureValue(for: outputName) else {
+                completion(nil, NSError(domain: "RFClassificationModel", code: 3, userInfo: [NSLocalizedDescriptionKey: "Could not find model output"]))
+                return
+            }
+            
+            var predictions: [RFPrediction] = []
+            
+            // Handle different output types
+            if let multiArray = output.multiArrayValue {
+                // Multi-array output (may be logits that need softmax conversion)
+                let rawValues = multiArray.dataPointer.bindMemory(to: Float.self, capacity: multiArray.count)
+                
+                // Check if values are logits (outside 0-1 range) and need softmax
+                var needsSoftmax = false
+                for i in 0..<multiArray.count {
+                    if rawValues[i] < 0.0 || rawValues[i] > 1.0 {
+                        needsSoftmax = true
+                        break
+                    }
+                }
+                
+                let probabilities: [Float]
+                if needsSoftmax {
+                    // Apply softmax to convert logits to probabilities
+                    probabilities = applySoftmax(logits: rawValues, count: multiArray.count)
+                } else {
+                    // Values are already probabilities
+                    probabilities = Array(UnsafeBufferPointer(start: rawValues, count: multiArray.count))
```

Comment:
this doesnt seem to handle the multiclass classification case (would apply sigmoid instead of softmax)

Reply:
I add the sigmoid to the graph if its multiclass in which case the sum should equal 1 right?; https://github.com/roboflow/roboflow-train/blob/97f60d9b6b8bab6d2186d6476783126a206cd6a8/train/src/classification/classification_trainer.py#L318C5-L363C39

Reply:
updated to use value passed from training, sigmoid is included in model as before